src/evict/evict_lru.c

```diff
@@ -1874,6 +1874,24 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
         WT_STAT_DATA_INCR(session, cache_eviction_walk_saved_pos);
     }
 
+#define FIND_RANDOM_PAGE_IF_NONE()                                                    \
+    if (btree->evict_ref == NULL) {                                                   \
+        for (;;) {                                                                    \
+            WT_WITH_PAGE_INDEX(session,                                               \
+              ret = __wt_random_descent(session, &btree->evict_ref,                   \
+                WT_READ_CACHE | WT_READ_NO_EVICT | WT_READ_NO_GEN | WT_READ_NO_WAIT | \
+                  WT_READ_NOTFOUND_OK | WT_READ_RESTART_OK,                           \
+                &session->rnd));                                                      \
+            if (ret != WT_RESTART)                                                    \
+                break;                                                                \
+            WT_STAT_CONN_INCR(session, cache_eviction_walk_restart);                  \
+            WT_STAT_DATA_INCR(session, cache_eviction_walk_restart);                  \
+        }                                                                             \
+        WT_RET_NOTFOUND_OK(ret);                                                      \
+    }
+
+    FIND_RANDOM_PAGE_IF_NONE();
```

Comment:
Can you add some comments here to explain the scenario that this helps to improve?

Reply:
Done

Reply:
Is this the change same as:

```
   if (btree->evict_ref == NULL)
          btree->evict_start_type = WT_EVICT_WALK_RAND_NEXT;
```

Reply:
Not exactly.
1. If random_descent returns NULL, then the search will be repeated if it's a random strategy.
2. `evict_start_type` stays the same.
But you are right that linear and random strategies do the same thing now.
This can change when we start using soft pointers.

Reply:
Can you add a comment to explain this?